### PR TITLE
Oracle list UI

### PR DESCRIPTION
**Pull request by mikec**

### CODE REVIEW

File: test/OracleManagerApp.test.js [hunk 1]
```diff
@@ -47,64 +47,69 @@ contract('OracleManagerApp', (accounts) => {
...
-      expect(logs[0].args.dataFeedAddress).to.equal(dataFeed1.address)
-      expect(logs[1].args.dataFeedAddress).to.equal(dataFeed2.address)
-    })
-
-    it('reverts if duplicate dataFeeds are submitted', async () => {
-      return assertRevert(async () => {
-        await oracleManagerApp.initialize([dataFeed1.address, dataFeed2.address, dataFeed1.address], medianDataFeed.address)
-      })
-    })
+    // Renable adding data feeds on initialize when
```
Comment (emaG3m) [inline test/OracleManagerApp.test.js]: would be awesome to have a `TODO:` here. convenient in later stages to see how many `TODO`'s we have left

File: app/actions/index.js [hunk 2]
```diff
@@ -2,11 +2,19 @@ import _ from 'lodash'
 import { logDebug, logError } from '../util/logger'
 import client from '../client'
 
-export const addOracle = ({ address }) => dispatch => {
-  return client.addOracle({ address }).then(txHash => {
-    logDebug(`client.addOracle: tx:`, txHash)
+export const addDataFeed = ({ address }) => dispatch => {
+  return client.addDataFeed({ address }).then(txHash => {
+    logDebug(`client.addDataFeed: tx:`, txHash)
```
Comment (emaG3m) [inline app/actions/index.js]: do these only log if the app is running in a particular debug mode

File: app/aragonRedux/aragonRedux.js [hunk 3]
```diff
@@ -1,7 +1,10 @@
 import _ from 'lodash'
 
-export const differenceWithEvents = (initialEventState, finalEventState) => {
-  return _.differenceWith(finalEventState, initialEventState, (a, b) => {
+export const filterNewEvents = (initialEventState, finalEventState) => {
```
Comment (emaG3m) [inline app/aragonRedux/aragonRedux.js]: what is this for?

File: test/OracleManagerApp.test.js [hunk 4]
```diff
@@ -47,64 +47,69 @@ contract('OracleManagerApp', (accounts) => {
     oracleManagerApp = await OracleManagerApp.at(receipt.logs.filter(l => l.event == 'NewAppProxy')[0].args.proxy)
     const MANAGE_DATA_FEEDS = await oracleManagerApp.MANAGE_DATA_FEEDS()
     await acl.createPermission(root, oracleManagerApp.address, MANAGE_DATA_FEEDS, root);
-    //
+    // //
```
Comment (emaG3m) [inline test/OracleManagerApp.test.js]: ??

File: app/actions/index.js (hunk 2 above)
Comment (mikec) [reply app/actions/index.js]: yep, we have this in futarchy-app as well

File: app/aragonRedux/aragonRedux.js (hunk 3 above)
Comment (mikec) [reply app/aragonRedux/aragonRedux.js]: aragon's event observable subscribes sometimes give the same event more than once, so this checks the tx hash to filter out any events that our reducer's have already seen

**Comment by mikec** [comment]
@emaG3m I addressed those comments

File: app/aragonRedux/aragonRedux.js (hunk 3 above)
Comment (emaG3m) [reply app/aragonRedux/aragonRedux.js]: lol

**Comment by emaG3m** [comment]
👍

**Pull request merged**